hm_1/tasks/task_02_02.py

```diff
@@ -0,0 +1,20 @@
+"""
+Given an array of size n, find the most common and the least common elements.
+The most common element is the element that appears more than n // 2 times.
+The least common element is the element that appears fewer than other.
+You may assume that the array is non-empty and the most common element
+always exist in the array.
+Example 1:
+Input: [3,2,3]
+Output: 3, 2
+Example 2:
+Input: [2,2,1,1,1,2,2]
+Output: 2, 1
+"""
+from collections import Counter
+from typing import List, Tuple
+
+
+def major_and_minor_elem(inp: List) -> Tuple[int, int]:
+    inp = Counter(inp).most_common()
```

Comment:
What is the complexity of this code?

Reply:
From documentation it’s O(n*logn)

Reply:
Can we do it with better complexity?

Reply:
I think no, because without it, 
I succeeded to achieve O(n^2) but it's more than O(nlogn)

Reply:
> I succeeded to achieve O(n^2)

How?

Reply:
def major_and_minor_elem(inp: List) -> Tuple[int, int]:
    #inp = Counter(inp).most_common()
    d ={}
    for i in set(inp):
        d[i] = inp.count(i) - this code has O(n^2)
    return max(d, key=d.get),min(d, key=d.get)- and this O(n)
    #return inp[0][0], inp[len(inp) - 1][0]--- =>    O(n^2+n)~ O(n^2)

Reply:
Please format code as code 
![image](https://user-images.githubusercontent.com/171597/111528389-244d4f80-8772-11eb-9448-8530cc8197e6.png)

So we need to make this part as O(n) to achieve total O(n).
An this can be done by changing one line and you already used that approach before.

```
d ={}
for i in set(inp):
    d[i] = inp.count(i) - this code has O(n^2)
```